pkg/util/util.go

```diff
@@ -99,6 +106,27 @@ func NamespaceKubernetesObject(componentName string, applicationName string) (st
 	return fmt.Sprintf("%s-%s", strings.Replace(componentName, "/", "-", -1), applicationName), nil
 }
 
+func GetRelGlobExps(directory string, globExps []string) []string {
+	relGlobExps := []string{}
+	for _, globExp := range globExps {
+		// for glob matching with the library
+		// the relative paths in the glob expressions need to be converted to absolute paths
+		var rel string
+		var err error
+		globExp = strings.ReplaceAll(globExp, "*", "_starchar_")
```

Comment:
If we need to use such a hack, it is because we didn't understand how to use the library, or it is buggy, or ...

Could you please create a new PR with the other gitignore library, so we can see which one is the best in our case?

Reply:
The problem is filepath.Rel does not calculate relative path when there are symbols like *